cardinal/world.go

```diff
@@ -296,11 +326,15 @@ func (w *World) Tick(ctx context.Context) error {
 	return nil
 }
 
+func (w *World) GetMessageManager() *message.Manager {
```

Comment:
Yeah, I know. I'm fully aware of this trade off. Here's the thing, I can't think of any other way to make this work with this signature:

`func GetMessage[K, V](wCtx)`

The problem is it needs to access a private var but GetMessage is generic and it can't be a method so it can't get access to the internal var that I need. Thus the only way is to make certain things public. 

This PR is made small because I want people to see the tradeoff I made and discuss it before going further. Let me know what you think and if you have any ideas on how to approach this.

Reply:
see comments i left above, imo there is just cascading complexity explosion here that will just make your life worse.

also, i'm pretty sure you do `wCtx.world.msgManager` since it's both in the same package.

Reply:
left follow up comment on GetMessage below. copying it here for clarity

> nvm, i understand what you are trying to do now, you still can't use message.MessageType though, I would make this private so you don't shoot yourself in the foot. Instead, you need to use types.Message.

I would also move this to cardinal.go together with GetComponent